integrations/iceberg-views/src/test/java/org/apache/iceberg/nessie/TestNessieIcebergViews.java

```diff
@@ -83,7 +83,7 @@ public TestNessieIcebergViews() {
   public void beforeEach() throws Exception {
     try {
       api.commitMultipleOperations()
-          .branchName(branch)
+          .branch(reference)
```

Comment:
Without the fix, this now (rightfully) throws:

```
org.projectnessie.error.NessieBadRequestException: Bad Request (HTTP/400): commitMultipleOperations.expectedHash: must not be null

	at org.projectnessie.error.ErrorCode.lambda$asException$1(ErrorCode.java:66)
	at java.base/java.util.Optional.map(Optional.java:265)
	at org.projectnessie.error.ErrorCode.asException(ErrorCode.java:66)```

Reply:
IIRC we have a few (Spark) SQL tests that rely on being able to commit without specifying the expected hash.

Reply:
Where are those tests? I would like to take a closer look. In any case, we'll need to decide what to do, we cannot let them as they are now since they are probably relying on requests not being properly validated.

Reply:
Are you referring to the tests in `:nessie-spark-extensions`? They seem to be passing.

Reply:
Hmm I see you are probably referring to NesQuEIT tests. Some of them are failing indeed with the error above. I will take a look.

Reply:
I am not sure I agree: actual production servers (Quarkus-based) have been validating this since the beginning. Only JAX-RS tests, to my best knowledge, were missing this validation and being a little too permissive.

Reply:
Ah, sorry, I missed that part :facepalm:

Reply:
So, I double-checked NesQuEIT tests and it appears they are passing fine with this PR. They will become problematic for my other PR #7396 , but here I think we are good.